ethcore/src/blockchain.rs

```diff
@@ -23,6 +23,24 @@ use extras::*;
 use transaction::*;
 use views::*;
 
+/// Uniquely identifies block.
+pub enum BlockId {
+	/// Block's sha3.
+	/// Querying by hash is always faster.
+	Hash(H256),
+	/// Block number within canon blockchain.
+	Number(BlockNumber)
+}
+
+/// Uniquely identifies transaction.
+pub enum TransactionId {
+	/// Transaction's sha3.
+	Hash(H256),
+	/// Block id and transaction index within this block.
+	/// Querying by block position is always faster.
+	BlockPosition(BlockId, usize)
```

Comment:
Try to avoid using two words when one will do - `Position` or `Location` would be fine here.